docs/blog: Add blogpost for Unikraft Hacktoberfest '22 (#132)
unikraft/docs: content/en/blog/2022-10-08-unikraft-hacktoberfest.md

+++
title = "Unikraft Hacktoberfest '22"
date = "2022-10-08T09:00:00+01:00"
author = "Radu Nichita"
tags = ["hacktoberfest"]
+++

> ### A new beginning

Reviewer: ```suggestion
### And Another One
```

> Last weekend, Unikraft hosted a community event for the `Hacktoberfest` series.

Reviewer: ```suggestion
Last weekend, we hosted yet another Unikraft hackathon, part of [Hacktoberfest 2022](https://hacktoberfest.com/).
```

> The event was held in a hybrid way: the newcomers and some familiar faces met physically at `University POLITEHNICA of Bucharest`, whereas the other folks joined remotely on the `monkey-business` channel.

Reviewer: ```suggestion
This was a hybrid event: the newcomers and some familiar faces met physically at University POLITEHNICA of Bucharest, whereas the other folks joined remotely on [the Unikraft Discord server](http://bit.ly/UnikraftDiscord).
```

> ### First steps

Reviewer: ```suggestion
### First Steps
```

Firstly, there was a presentation given by [Stefan Jumărea](https://github.com/StefanJum), our community lead for documentation.
It was an introduction to unikernels and Unikraft and a brief presentation of hackathon challenges.

> ### Hardwork always pays off

Reviewer: ```suggestion
### Hard work Always Pays Off
```

> Afterwards, the participants dove deep into the chosen tasks.

Reviewer: ```suggestion
Afterwards, the participants dove deep into the chosen tasks, from the [`Hackathons` project on GitHub](https://github.com/orgs/unikraft/projects/29).
```

> Most of them started by running the ` application on top of Unikraft.

Reviewer: ```suggestion
Most of them started by running adding new [static PIE Linux executables](https://github.com/unikraft/static-pie-apps) to be run on top of Unikraft in binary compatibility mode using the [ELF Loader](https://github.com/unikraft/run-app-elfloader).
```

Then, they followed some tutorials from the `USOC '22` to get accustomed to the core components of Unikraft.

> After a few hours of intense hacking, results were on the table: 3 applications were ported as `static-build-pie` for Unikraft!!

Reviewer: ```suggestion
After a few hours of intense hacking, results were on the table: 3 applications were ported as static PIE apps for Unikraft!
And participants are working forward to port applications and submit PRs.
Another PRs was submitted the next day.
```

+++
title = "Unikraft Hacktoberfest '22"
date = "2022-10-08T09:00:00+01:00"

> author = "Radu Nichita"

Reviewer: ```suggestion
author = "Radu Nichita, Andrei Mutu"
```

tags = ["hacktoberfest"]
+++

### And Another One

> As tradition dictates, if it's Saturday then it is definitely another hackathon we are proud to host at Unikraft. Even so, the last one was a bit special. How you ask? Well, as we previously announced, we have been officially accepted to be part of the [Hacktoberfest 2022](https://hacktoberfest.com/).

Reviewer: ```suggestion
As tradition dictates, if it's Saturday then it is definitely another hackathon we are proud to host at Unikraft.
Even so, the last one was a bit special.
How you ask?
Well, as we previously announced, we have been officially accepted to be part of the [Hacktoberfest 2022](https://hacktoberfest.com/).
```

+++
title = "Unikraft Hacktoberfest '22"
date = "2022-10-08T09:00:00+01:00"
author = "Radu Nichita, Andrei Mutu"
tags = ["hacktoberfest"]
+++

### And Another One

As tradition dictates, if it's Saturday then it is definitely another hackathon we are proud to host at Unikraft.
Even so, the last one was a bit special.
How you ask?
Well, as we previously announced, we have been officially accepted to be part of the [Hacktoberfest 2022](https://hacktoberfest.com/).

Thus `UNIKRAFT HACKTOBERFEST '22` was born in the shape of a hybrid event.
Here both familiar faces and newcomers alike came together physically at University POLITEHNICA of Bucharest or joined remotely on [the Unikraft Discord server](http://bit.ly/UnikraftDiscord) in order to test their knowledge to unikernels.

### First Steps

In order to break the ice, our documentation community lead, [Stefan Jumărea](https://github.com/StefanJum), gave an introductory presentation.
This aimed to present both unikernels and Unikraft to the participants, introducing it to the newcomers and refreshing it for the veterans.

> Afterward, a short discussion about the challenges raised our participant's spirits.

Reviewer 2: I think here is a typo. It should be `participants` without the apostrophe.

tags = ["hacktoberfest"]
+++

### And Another One

> As tradition dictates if it's Saturday then it is definitely another hackathon we are proud to host at Unikraft. Even so, the last one was a bit special. How you ask? Well, as we previously announced, we have been officially accepted to be part of the [Hacktoberfest 2022](https://hacktoberfest.com/).

Author: ```suggestion
As tradition dictates, if it's Saturday then it is definitely another hackathon we are proud to host at Unikraft. Even so, the last one was a bit special. How you ask? Well, as we previously announced, we have been officially accepted to be part of the [Hacktoberfest 2022](https://hacktoberfest.com/).
```

Thus `UNIKRAFT HACKTOBERFEST '22` was born in the shape of a hybrid event. Here both familiar faces and newcomers alike came together physically at University POLITEHNICA of Bucharest or joined remotely on [the Unikraft Discord server](http://bit.ly/UnikraftDiscord) in order to test their knowledge in unikernels.

### First Steps

> In order to break the ice, our documentation community lead, [Stefan Jumărea](https://github.com/StefanJum), gave a introductory presentation. This aimed to present both unikernels and Unikraft to the participants, introducing it to the newcomers and refreshing it for the veterans. Afterwards, a short discussion about the challenges raised our participants spirits. Onto krafting young padawans!

Author: ```suggestion
In order to break the ice, our documentation community lead, [Stefan Jumărea](https://github.com/StefanJum), gave an introductory presentation. This aimed to present both unikernels and Unikraft to the participants, introducing it to the newcomers and refreshing it for the veterans. Afterward, a short discussion about the challenges raised our participant's spirits. Onto krafting young padawans!
```

### Hardwork Always Pays Off

It was finally time for the moment we have been waiting for, the deep dive into the tasks. In order to warm up and close the gap between traditional applications and Unikraft, most of the participants chose to run static PIE Linux executables](https://github.com/unikraft/static-pie-apps) on top of Unikraft in binary compatibility mode using the [ELF Loader](https://github.com/unikraft/run-app-elfloader).
The more experienced ones adventured even deeper and added new ones.

In their aid [`USOC '22`](https://unikraft.org/community/hackathons/usoc22/) tutorials also played a great role, allowing participants to have a better understanding of the core components of Unikraft.

> While begginings can sometimes be hard and unrewarding, this was not the case for our contestans. After a few hours the results were on the table: 3 applications were ported as static PIE apps for Unikraft.

Author: ```suggestion
While beginnings can sometimes be challenging and unrewarding, this was not the case for our contestants. After a few hours, the results were on the table: 3 applications were ported as static PIE apps for Unikraft.

> A great start! But wait, there's more! The participants continued working on their porting and submited their PRs. To our surprise we even had some PRs submited the next day.

Author: ```suggestion
A great start! But wait, there's more! The participants continued working on their porting and submitted their PRs. To our surprise we even had some PRs submitted even the next day!.
```
